### ISSUE TITLE
Jenkins Operator Active Directory

### ISSUE THREAD

**Issue by Chris-Carmichael** [jenkinsci/kubernetes-operator#642]
**Describe the bug**
I can't get the `active-directory` plugin to work.  I see an issue that was closed, [#47](https://github.com/jenkinsci/kubernetes-operator/issues/47) .    The last comment is pointing to a `security.md` file that no longer exists.

**To Reproduce**
Install `active-directory` plugin.  CasC fails to config anything and you are dead in the water.

**Additional information**

Kubernetes version:  1.20.x
Jenkins Operator version:  0.6.0

I think the issue is already known, but not solved.  Once AD auth takes over, the operator can't auth to config the pod.  I tried the workaround outlined in #47 of changing the secret to match an AD cred, but it didn't work.

Just trying to figure out if this is on the roadmap.  This project is rad, but without AD auth, it's a no-go for the company I currently work for.

**Comment by prryb** [comment]
Hello, here's a working link to Security: https://jenkinsci.github.io/kubernetes-operator/docs/getting-started/latest/security/

Are you experiencing the same 403 error as in #47?

The Operator tries to authenticate with Jenkins using the `jenkins-operator-credentials-<cr_name>` secret that it creates. You need to create an account for the operator in your AD and configure it to have admin permissions. https://github.com/jenkinsci/kubernetes-operator/issues/133 is quite dated but has some details that worked.

**Comment by Chris-Carmichael** [comment]
Yep, same error.  I tried to change the `user` and `password` keys in the secret, but it still didn't work. 

I was using the `active-directory` plugin, where I see the workaround that is documented is using the `ldap` plugin.  I'll try that instead and see what happens.

If this could get ironed out, I suspect you'll see more adoption from Jenkins admins that are accustomed to `active-directory`.  Also, documentation is sorely needed.  Finding out that this gap exists after spending time getting everything else working is a downer.  Still though...awesome project.  Just not production ready.

**Comment by prryb** [comment]
Looking at the code (https://github.com/jenkinsci/kubernetes-operator/blob/126b02aaab2088a052c638842c7c31e0c67f21d9/pkg/configuration/configuration.go#L250-L278), if the token was previously generated and is present in the `jenkins-operator-credentials-<cr_name>` when being modified, you could also try to remove `tokenCreationTime` to force token regeneration. `user` and `password` combo looks to be only used to generate the token. Later communication uses `BasicAuth` with `user` and `token`.

There are some major changes coming soon to this project. With that being said, this is still OSS, contributions are welcome to improve the documentation. For now, I would hold off with functional changes to the code till the new API appears in the project. 🙂

**Comment by Chris-Carmichael** [comment]
Oh that's great news to hear.  So excited.  Today was a bear, but I'll test that suggestion tomorrow for sure.

Thanks for being awesome.
